processing/src/main/java/io/druid/segment/IndexMaker.java

```diff
@@ -540,15 +554,9 @@ private static File makeIndexFiles(
 
     progress.progress();
     makeIndexBinary(
-        v9Smoosher,
-        adapters,
-        outDir,
-        mergedDimensions,
-        mergedMetrics,
-        skippedDimensions,
-        progress,
-        indexSpec
+        v9Smoosher, adapters, outDir, mergedDimensions, mergedMetrics, skippedDimensions, progress, indexSpec
     );
+    makeMetadataBinary(v9Smoosher, progress, commitMetaData);
```

Comment:
changed the api to work with Map<String, Object> segment metadata,
key "commitMeta" is supplied from RealtimePlumber
code here just works with Map object